GoogleForCreators/web-stories-wp
# Added focus management to trim component

## Context

This adds proper focus management to trim component including RTL support, which actually doesn't require any work, as the direction of a playing tape is universally left-to-right regardless of reading direction.

## To-do

* [x] Focus trapping
* [x] Auto-focus cancel button on mount
* ~Display sliders and scrims correctly in RTL mode~
* ~Handle moving sliders correctly when using keyboard in RTL mode~
* ~Handle moving sliders correctly when using mouse in RTL mode~

## User-facing changes

LTR:
![trim-focus-ltr](https://user-images.githubusercontent.com/637548/135365720-0fa48a04-399e-4892-bc82-1d3bd08dc735.gif)

RTL:
![trim-focus-rtl](https://user-images.githubusercontent.com/637548/135365668-ec9bcbe7-e257-425b-b7b4-cea4278355ce.gif)

## Testing Instructions

This PR can be tested by following these steps:

1. Add a video and enter trim mode
2. Observe how keyboard focus is immediately placed on the cancel button
3. Observe how tabbing keeps focus within the trim component
4. Observe how tab order is logical (cancel ➡️  start-handle ➡️  end-handle  ➡️  trim button ➡️  cancel) rather than physical (from one side of the screen to the other)
5. Observe how arrow keys move the handles correctly
6. Observe how the video play direction in the trim rail is always left-to-right as it follows the direction of a tape being played and not the passing of time ([as per Material Design guidelines](https://material.io/design/usability/bidirectionality.html#:~:text=Local%20cafe%20icon-,the%20passage%20of%20time,-Anything%20depicting%20the))
7. 6. Do the above in both RTL and LTR mode

## Checklist




---

Fixes #9180
